tests/testthat/test-all.R

```diff
@@ -0,0 +1,203 @@
+  df <- read.socrata('https://soda.demo.socrata.com/resource/4334-bgaj.csv')
+  expect_equal(1007, nrow(df), "rows")
+  expect_equal(9, ncol(df), "columns")
+})
+
+test_that("read Socrata JSON", {
+  df <- read.socrata('https://soda.demo.socrata.com/resource/4334-bgaj.json')
+  expect_equal(1007, nrow(df), "rows")
+  expect_equal(11, ncol(df), "columns")
+})
+
+test_that("read Socrata No Scheme", {
+  expect_error(read.socrata('soda.demo.socrata.com/resource/4334-bgaj.csv'))
+})
+
+test_that("readSoQL", {
+  df <- read.socrata('http://soda.demo.socrata.com/resource/4334-bgaj.csv?$select=region')
+  expect_equal(1007, nrow(df), "rows")
+  expect_equal(1, ncol(df), "columns")
+})
+
+test_that("readSoQLColumnNotFound (will fail)", {
+  # SoQL API uses field names, not human names
+  expect_error(read.socrata('http://soda.demo.socrata.com/resource/4334-bgaj.csv?$select=Region'))
+})
+
+test_that("URL is private (Unauthorized) (will fail)", {
+  expect_error(read.socrata('http://data.cityofchicago.org/resource/j8vp-2qpg.json'))
+})
+
+test_that("readSocrataHumanReadable", {
+  df <- read.socrata('https://soda.demo.socrata.com/dataset/USGS-Earthquake-Reports/4334-bgaj')
+  expect_equal(1007, nrow(df), "rows")
+  expect_equal(9, ncol(df), "columns")
+})
+
+test_that("format is not supported", {
+  # Unsupported data formats
+  expect_error(read.socrata('http://soda.demo.socrata.com/resource/4334-bgaj.xml'))
+})
+
+context("Socrata Calendar")
+
+test_that("Calendar Date Long", {
+  df <- read.socrata('http://soda.demo.socrata.com/resource/4334-bgaj.csv')
+  dt <- df$Datetime[1] # "2012-09-14 22:38:01"
+  expect_equal("POSIXlt", class(dt)[1], "data type of a date")
+  expect_equal(2012, dt$year + 1900, "year")
+  expect_equal(9, dt$mon + 1, "month")
+  expect_equal(14, dt$mday, "day")
+  expect_equal(22, dt$hour, "hours")
+  expect_equal(38, dt$min, "minutes")
+  expect_equal(1, dt$sec, "seconds")
+})
+
+test_that("Calendar Date Short", {
+  df <- read.socrata('http://data.cityofchicago.org/resource/y93d-d9e3.csv?$order=debarment_date')
+  dt <- df$DEBARMENT.DATE[1] # "05/21/1981"
+  expect_equal("POSIXlt", class(dt)[1], "data type of a date")
+  expect_equal(81, dt$year, "year")
+  expect_equal(5, dt$mon + 1, "month")
+  expect_equal(21, dt$mday, "day")
+  expect_equal(0, dt$hour, "hours")
+  expect_equal(0, dt$min, "minutes")
+  expect_equal(0, dt$sec, "seconds")
+})
+
+context("Checks the validity of 4x4")
+
+test_that("is 4x4", {
+  expect_true(isFourByFour("4334-bgaj"), "ok")
+  expect_false(isFourByFour("4334c-bgajc"), "11 characters instead of 9")
+  expect_false(isFourByFour("433-bga"), "7 characters instead of 9")
+  expect_false(isFourByFour("433-bgaj"), "3 characters before dash instead of 4")
+  expect_false(isFourByFour("4334-!gaj"), "non-alphanumeric character")
+})
+
+
+test_that("is 4x4 URL", {
+  expect_error(read.socrata("https://soda.demo.socrata.com/api/views/4334c-bgajc"), "11 characters instead of 9")
```

Comment:
I see the issue, just some differences in the RUnit approach vesus testthat. For instance, this test passes with the syntax:
```r
test_that("is 4x4 URL", {
  expect_error(read.socrata("https://soda.demo.socrata.com/api/views/4334c-bgajc"), "4334c-bgajc is not a valid Socrata dataset unique identifier", label="11 characters instead of 9")
  expect_error(read.socrata("https://soda.demo.socrata.com/api/views/433-bga"), "433-bga is not a valid Socrata dataset unique identifier", label="7 characters instead of 9")
  expect_error(read.socrata("https://soda.demo.socrata.com/api/views/433-bgaj"), "433-bgaj is not a valid Socrata dataset unique identifier", label="3 characters before dash instead of 4")
  expect_error(read.socrata("https://soda.demo.socrata.com/api/views/4334-!gaj"), "4334-!gaj is not a valid Socrata dataset unique identifier", label="non-alphanumeric character")
})
```
I will make changes to the tests that fail.